lib/telemetry_registry.ex

```diff
@@ -0,0 +1,260 @@
+defmodule TelemetryRegistry do
+  @moduledoc """
+  TelemetryRegistry provides tools for the discovery and documentation of [telemetry](https://github.com/beam-telemetry/telemetry)
+  events within your applications.
+
+  ## Telemetry Event Definitions and Declaration
+
+  Users want to know what telemetry events are available in your library, what they mean, as well as what
+  the measurements and metadata maps contain. TelemetryRegistry creates an official standard and mechanism
+  for telemetry event declaration and definition.
+
+  ### Who Should Document Events?
+
+  Library authors in particular should provide event declarations for documentation and to simplify tracing.
+  Of course, everyone should document their modules!
+
+  ### Where Should I Declare Events?
+
+  Events should only be declared once, usually in the module from which it originates. Event names should _always_
+  be namespaced to your application or library. For example, if your application is an http client, your events
+  should start with the name of your application, not a generic name.
+
+  **Do:** `[:tesla, :request, :stop]`
+
+  **Don't:** `[:http_client, :request, :stop]`
+
+  ### Event Definition Format
+
+  Events are declared using the `telemetry_event` module attribute. The attribute accepts an event definition
+  which are used for producing documentation and event discovery. All definition keys are required.
+
+  ```elixir
+  %{
+    event: [:my_app, :event, :stop],
+    description: "A description of what the event is and when it is emitted",
+    measurements: "A string containing a pseudo or typespec - see examples",
+    metadata: "A string containing a pseudo or real typespec - see examples"
+  }
+  ```
+
+  ```erlang
+  \#{
+    event => [my_app, event, stop],
+    description => <<"A description of what the event is and when it is emitted">>,
+    measurements => <<"A string containing a pseudo or typespec - see examples">>,
+    metadata => <<"A string containing a pseudo or real typespec - see examples">>
+  }
+  ```
```

Comment:
Do we need to have Erlang examples in the Elixir docs? 🤔 Maybe this should be moved to the README?

Reply:
As there will be only one docs published, and there is no support for Erlang docs in ExDocs then I think it would be nice to have such information there, at least for now, until EEP48 fully lands.

Reply:
I'll defer to @hauleth on this. What's the timeframe on that? This year?